components/Modals/SignUp.tsx

```diff
@@ -4,40 +4,59 @@ import Avatar from 'components/Avatar'
 import { useState, useEffect } from 'react'
 import Modal from 'react-modal'
 import CloseIcon from 'public/assets/icons/close.svg'
-import { CopyButton } from 'shared'
+import { CopyButton, Loader, RadioButton, RadioGroup } from 'shared'
 import HorizontalScrollView from 'components/HorizontalScrollView'
-import {
-  loginUser,
-  setUserAvatar,
-  setUserRegistered,
-  createUser,
-} from 'lib/user'
-import { useUser, useHasMounted, useTranslations, useLang } from 'hooks'
+import { useHasMounted, useTranslations, useLang } from 'hooks'
 import clsx from 'clsx'
+import { useAuthContext } from 'providers/AuthProvider'
+import { generateKeypair, generateKeyPairFromString } from 'lib/crypto'
+import { register } from 'api/auth'
+import { Input } from 'shared'
+import { PrivateKey } from 'types'
 
-export default function SignUpModal({ open, onClose, onConfirm }) {
+enum View {
+  Generate = 'generate',
+  Input = 'input',
+}
+
+export default function SignUpModal({ open, onClose }) {
   const lang = useLang()
   const t = useTranslations(lang)
-  const hasMounted = useHasMounted()
-  const { user } = useUser()
-  let [avatar, setAvatar] = useState(1)
 
-  function saveLocally() {
-    setUserAvatar(avatar)
-    loginUser()
+  const [loading, setLoading] = useState<boolean>(false)
+  const [avatar, setAvatar] = useState(1)
+  const [view, setView] = useState<string>(View.Generate)
+  const [privateKey, setPrivateKey] = useState<PrivateKey | undefined>(
+    undefined
+  )
+
+  const handleSetPrivateKey = (pk: string) => {
+    const { sec } = generateKeyPairFromString(pk)
+
+    setPrivateKey(sec)
   }
 
-  function confirm() {
-    saveLocally()
-    setUserRegistered()
-    onConfirm()
+  async function handleConfirm() {
+    try {
+      setLoading(true)
+
+      await register(privateKey, `/assets/avatars/${avatar}.png`)
+
+      onClose()
```

Comment:
With this happening on every create account button press is there a concern about account generation spam? 
Even just trying it out I think I made several accounts as I didn't copy my private key the first time

Reply:
Yeah that's a valid concern, something I wanted to save for a follow up PR but a few things we could do are:
- Add rate limiting on backend
- Add a captcha